crates/contract/src/data_transfer_object_mapping.rs

```diff
@@ -0,0 +1,157 @@
+use attestation::{
+    attestation::{Attestation, DstackAttestation, MockAttestation},
+    collateral::{Collateral, QuoteCollateralV3},
+    EventLog, TcbInfo,
+};
+use data_transfer_objects::dto_attestation::{
+    DtoAttestation, DtoCollateral, DtoDstackAttestation, DtoEventLog, DtoMockAttestation,
+    DtoTcbInfo,
+};
+
+pub(crate) trait ConvertDtoToContractType {
```

Comment:
Two things:

1. This should be a public trait right?
2. Nit: Naming suggestion: I think `IntoContractType` is more concise and conveys the same information. Open to other suggestions, but having `Dto` as part of the name here sounds weird.

```suggestion
pub trait IntoContractType {
```

Reply:
1. No, only the contract should have access to these mappings. At some point the contract should be a standalone binary, and not a library we depend from other crates. So everything in that crate should not need to be more public than `pub(crate)`.

2. Good point, that's indeed a better name. https://github.com/near/mpc/pull/1151/commits/cfd7e87e267113fe840fd7ea6eceac00afbd876d